extensions/amp-subscriptions/0.1/amp-subscriptions.js

```diff
@@ -302,13 +302,14 @@ export class SubscriptionService {
             Entitlement.empty('local'));
       }
       const decodedData = this.jwtHelper_.decode(authData);
-      user().assert(
-          decodedData['aud'] == origin ||
-          decodedData['aud'] == sourceOrigin,
-          `The mismatching "aud" field: ${decodedData['aud']}`);
-      // Expiration time is in seconds.
-      user().assert(decodedData['exp'] > Math.floor(Date.now() / 1000),
-          'Payload is expired');
+
+      if (decodedData['aud'] != origin || decodedData['aud'] == sourceOrigin) {
```

Comment:
Can we instead create a `verifyAuthResponse` method that'd return promise and process it's signals?

E..g.
```
this.verifyAuth(authData).then(entitlement => {
  this.platformStore_.resolveEntitlement('local', entitlement);
  // whatever lese
}, reason => {
  this.viewer_.sendMessage('auth-rejected', dict({
      'reason': String(reason),
    }));
});

verifyAuth() {
  return new Promise(resolve => {
    // Parse JWT
    // Verify JWT
    // Convert to entitlement
    resolve(entitltement);
  });
}
```

This is a bit cleaner and when we add cache validation, most of this won't change.

Reply:
I can do this, create a new class may be, convert each phase(parse, verify, convert) into its own function with cache validation and for now unblock the N360 integration